frontend/src/pages/user/home/Words/hooks/useWords.js

```diff
@@ -0,0 +1,51 @@
+import { useState, useEffect } from "react";
+import { useNavigate } from "react-router-dom";
+import { useDispatch, useSelector } from "react-redux";
+
+import { authenticate } from "../../../../../helpers/auth";
+import { submitAnswer } from "../../../../../helpers/api";
+
+export const useWords = (lesson) => {
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+  const { user } = useSelector((state) => state.currentUser);
+  const { currentLesson } = useSelector((state) => state.lesson);
+  const [lessonWords, setLessonWords] = useState([]);
+  const [currentQuestion, setCurrentQuestion] = useState(null);
+  const [currentNumber, setCurrentNumber] = useState(0);
```

Comment:
- although this would work. But I think better approach of this one is to move these states and actions (`nextQuestion`, `saveAnswer`, `submitAnswers`, etc.) to reducer instead of hooks, so we can access the states globally anywhere (this approach is like from our previous project. fyi).